server/src/main/java/com/linecorp/line/auth/fido/fido2/server/util/ReqMsgVerifier.java

```diff
@@ -1,181 +0,0 @@
-        }
-
-        // serverPublicKeyCredential.response
-        if (StringUtils.isEmpty(verifyCredential.getServerPublicKeyCredential().getResponse().getAuthenticatorData())) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "serverPublicKeyCredential.response.authenticatorData should not be null or empty");
-        }
-
-        if (StringUtils.isEmpty(verifyCredential.getServerPublicKeyCredential().getResponse().getSignature())) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "serverPublicKeyCredential.response.signature should not be null or empty");
-        }
-
-        if (StringUtils.isEmpty(verifyCredential.getServerPublicKeyCredential().getResponse().getClientDataJSON())) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "serverPublicKeyCredential.response.clientDataJSON should not be null or empty");
-        }
-
-        try {
-            Base64.getUrlDecoder().decode(verifyCredential.getServerPublicKeyCredential().getResponse().getAuthenticatorData());
-        } catch (IllegalArgumentException e) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "serverPublicKeyCredential.response.authenticatorData should be base64 url encoded");
-        }
-
-        try {
-            Base64.getUrlDecoder().decode(verifyCredential.getServerPublicKeyCredential().getResponse().getSignature());
-        } catch (IllegalArgumentException e) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "serverPublicKeyCredential.response.signature should be base64 url encoded");
-        }
-
-        try {
-            Base64.getUrlDecoder().decode(verifyCredential.getServerPublicKeyCredential().getResponse().getClientDataJSON());
-        } catch (IllegalArgumentException e) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "serverPublicKeyCredential.response.clientDataJSON should be base64 url encoded");
-        }
-    }
-
-    public static void validateRegOptionRequest(RegOptionRequest regOptionRequest) {
-        // validation
-        if (regOptionRequest.getRp() == null ||
-                regOptionRequest.getUser() == null) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY);
-        }
-
-        if (regOptionRequest.getUser().getId() == null) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "userId should not be null");
-        }
-        if (!(regOptionRequest.getUser().getId().length() > 0 && regOptionRequest.getUser().getId().length() < 64)) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.INVALID_REQUEST_BODY, "userId length must be between 1 and 64 bytes");
```

Comment:
Is it correct that just 64 bytes are an error?

Reply:
Yes, that's right.
You can find the rules here.
https://www.w3.org/TR/webauthn/#sctn-createCredential
`If the length of options.user.id is not between 1 and 64 bytes (inclusive) then return a TypeError.`

BTW, I'm planning to proceed with the review soon. 
Thanks!

Reply:
Thank you for reviewing 😃

>If the length of options.user.id is not between 1 and 64 bytes (inclusive) then return a TypeError.

It looks like 64 bytes are not inclusive.
```java
regOptionRequest.getUser().getId().length() < 64
```
I may be wrong, but I think the following is correct.
```java
regOptionRequest.getUser().getId().length() <= 64
```

Reply:
Oh yes, I think your opinion is correct.

Reply:
https://github.com/line/line-fido2-server/pull/20/files#diff-f579719df9002ef23bdab8a28a4d4dc7986c3ceec90b9f809ad5fef98af6e94dR30

Ok, I will change the maximum length

`@Length(min = 1, max = 63)` -> `@Length(min = 1, max = 64)`